text/3453-f16-and-f128.md

```diff
@@ -0,0 +1,132 @@
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+While `f32` and `f64` have very broad support in most hardware, hardware support for `f16` and `f128` is more niche. On most systems software emulation will be required. Therefore, the main drawback is implementation difficulty.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There are some crates aiming for similar functionality:
+
+- [f128](https://github.com/jkarns275/f128) provides binding to `__float128` type in GCC.
+- [half](https://crates.io/crates/half) provides implementation of binary16 and bfloat16 types.
+
+However, besides the disadvantage of usage inconsistency between primitive type and type from crate, there are still issues around those bindings.
+
+The ability to accelerate additional float types depends on CPU/OS/ABI/features of different targets heavily. Evolution of LLVM may unlock possibility of accelerating the types on new targets. Implementing them in the compiler allows the compiler to perform optimizations for hardware with native support for these types.
+
+Crates may define their type on top of a C binding, but extended float type definition in C is complex and confusing. The meaning of C types may vary by target and/or compiler options. Implementing `f16` and `f128` in the Rust compiler helps to maintain a stable codegen interface and ensures that all users have one single canonical definition of 16-bit and 128-bit float types, making it easier to exchange data between crates and languages.
+
+# Prior art
+[prior-art]: #prior-art
+
+Many other languages and compilers have support for these proposed float types. As mentioned above, C has `_Float16` and `_Float128`, and C++ has `std::float16_t` and `std::float128_t`. Glibc supports 128-bit floats in software on [many architectures](https://sourceware.org/git/?p=glibc.git;a=blob;f=NEWS;hb=81325b12b14c44887f1633a2c180a413afc2b504#l143).
+
+This RFC is designed as a subset of [RFC 3451](https://github.com/rust-lang/rfcs/pull/3451), which proposes adding a variety of float types, including ones not in this RFC designed for interoperability with other languages.
+
+Both this RFC and 3451 are built upon the discussion in [issue 2629](https://github.com/rust-lang/rfcs/issues/2629).
+
+The main consensus of the discussion thus far is that more float types would be useful, especially the IEEE 754 types proposed in this RFC as `f16` and `f128`. Other types can be discussed in a future RFC.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+The main unresolved parts of this RFC are the implementation details in the context of the Rust compiler and standard library. The behavior of `f16` and `f128` is well-defined by the IEEE 754 standard, and is not up for debate. Whether these types should be included in the language is the main question of this RFC, which will be resolved when this RFC is accepted.
```

Comment:
You can delete the "should we accept this RFC" part from unresolved questions, it's innate and that discussion should be in rationale & alternatives (which it is)